snapcraft/plugins/nodejs.py

```diff
@@ -178,19 +178,27 @@ def build(self):
                 self._manifest["yarn-lock-contents"] = lock_file.read()
 
         # Get the names and versions of installed packages
-        installed_node_packages = self._get_installed_node_packages(self.installdir)
-        self._manifest["node-packages"] = [
-            "{}={}".format(name, installed_node_packages[name])
-            for name in installed_node_packages
-        ]
+        if self.options.nodejs_package_manager == "npm":
+            installed_node_packages = self._get_installed_node_packages(self.installdir)
+            self._manifest["node-packages"] = [
+                "{}={}".format(name, installed_node_packages[name])
+                for name in installed_node_packages
+            ]
+        # Skip this step if yarn is used, as it may produce different
```

Comment:
Can you expand on this?

Reply:
Basically, I've noticed that most builds that I'm doing with yarn will crash at this step in the build because `npm ls` returns an error after it can't find all the dependencies in the tree. Yarn structures its dependencies differently from npm, so it's common for the dependency tree for npm to be different from the one for yarn. Since we can't get the output of `yarn ls` easily, I just skip this step for yarn.